controllers/drplacementcontrol.go

```diff
@@ -785,6 +792,13 @@ func (d *DRPCInstance) updateUserPlacementRuleStatus(status plrv1.PlacementRuleS
 }
 
 func (d *DRPCInstance) createVRGManifestWork(homeCluster string) error {
+	err := d.ensureNamespaceExistsOnManagedCluster(homeCluster)
```

Comment:
Thinking... should we check for the existence of the VRG MW here before we use the ManagedClusterView to retrieved the Namespace.  If MW exists, then we know that the namespace exists.  If it doesn't exist, then we don't know and hence, we make this call.  Again, just a thought, as MCV is very expensive and it is guaranteed it will not complete in one shot.

Reply:
Another question, if let's say the MCV fails the first time getting the Namespace, with a failure other than NotFound.  The second time we get here, will it create a new MCV.  I doubt it.  That means there is a potential we get stuck here forever.  Is that also true for all the calls to getManagedClusterResource.  Hmm.... I just checked the code and I think we are in trouble here.  I think we should delete the MCV on any result so next time it will be recreated.  Again, just thinking.